infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,169 @@
+            exit 1
+        fi
+        read -p "Or enter VCH name to continue: " VIC_NAME
+    fi
+}
+
+# Falls back on VIC_ID if possible, bails if not, assumes the VCH name is ambiguous
+function check-name-isnt-ambiguous ()  {
+    username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
+    password=$(govc env | grep GOVC_PASSWORD | cut -d= -f2)
+    target=$(govc env | grep GOVC_URL | cut -d= -f2)
+
+    if [[ ! -v VIC_ID ]] && [[ $(vch-name-is-ambiguous) ]]; then
+        echo "The provided VIC name is ambiguous; please choose the correct VCH ID from the output below and assign it to the environment variable VIC_ID, e.g., export VIC_ID=12"
+
+        $VIC_DIR/bin/vic-machine-linux\
+            ls \
+            --target $GOVC_URL \
+            --user $GOVC_USER \
+            --password=$GOVC_PASSWORD \
+            --thumbprint=$(get-thumbprint)
+
+        if [[ $interactive -eq 0 ]]; then
+            exit 1
+        fi
+        read -p "Enter unique VIC ID from above: " VIC_ID
+    fi
+}
+
+# Translates VCH name to ID if necessary
+function get-vic-id () {
+    if [[ -z $VIC_ID ]]; then
+        export VIC_ID="$($VIC_DIR/bin/vic-machine-linux ls --target=$target --user=$username --password=$password --thumbprint=$(get-thumbprint) | grep $VIC_NAME | awk '{print $1}')"
+    fi
+}
+
+function get-ssh-keys() {
+    if [[ -z $VIC_KEY ]]; then
+        echo "Variable VIC_KEY not set. Provide the path to your public SSH key below."
+        if [[ $interactive -eq 0 ]]; then
+            exit 1
+        fi
+        read -p "Path to your public SSH key for access to VCH [/home/$USER/.ssh/id_rsa.pub]: " key
+        export VIC_KEY=${key:-/home/$USER/.ssh/id_rsa.pub}
+    fi
+}
+
+# Checks environment for required inputs
+function sanity-checks () {
+    check-govc-vars
+    check-vch-name-or-id
+    check-name-isnt-ambiguous
+    get-vic-id
+    get-ssh-keys
+}
+
+# Enables SSH and saves off the VCH IP address
+function enable-debug () {
+    VCH_IP=$($VIC_DIR/bin/vic-machine-linux debug \
+                                   --target=$target \
+                                   --id=$VIC_ID \
+                                   --user=$username \
+                                   --password=$password \
+                                   --authorized-key=$VIC_KEY \
+                                   --thumbprint=$(get-thumbprint) \
+                 | grep -A1 "Published ports" | tail -n1 | awk '{print $NF}')
+}
+
+# SCPs the component in $1 to the VCH, plops it in place, and brutally kills the previous running process
+function replace-component() {
+    scp -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no $VIC_DIR/bin/$1 root@$VCH_IP:/tmp/$1 2>/dev/null
+    pid=$(on-vch ps -e --format='pid,args' \
+                 | grep $1 | grep -v grep | awk '{print $1}')
+    on-vch chmod 755 /tmp/$1
+    on-vch mv /tmp/$1 /sbin/$1
+    if [[ $1 == "vic-init" ]]; then
+        on-vch systemctl restart vic-init
```

Comment:
We might need to look at the effect of this with #6943 - it's possible that this would cause all components to be shutdown as well which is not what we're after.